Products/DataCollector/ApplyDataMap/applydatamap.py

```diff
@@ -399,6 +403,9 @@ def _process_relationshipmap(relmap, base_device):
         _validate_datamap(parent, object_map)
         for object_map in relmap.maps
     ]
+    for map in new_maps:
+        map.plugin_name = relmap.plugin_name
```

Comment:
Although I'm sure this isn't currently happening, probably wise to wrap this in `if not map.plugin_name`

Reply:
We considered this, and determined that even if an object map somehow had a plugin_name that differed from the relationship map's, they should use the one from the relmap